pkg/codegen/pcl/binder_schema_test.go

```diff
@@ -19,3 +22,34 @@ func BenchmarkLoadPackage(b *testing.B) {
 		contract.AssertNoError(err)
 	}
 }
+
+func TestGenEnum(t *testing.T) {
+	t.Parallel()
+	enum := &model.EnumType{
+		Elements: []cty.Value{
+			cty.StringVal("foo"),
+			cty.StringVal("bar"),
+		},
+		Type:  model.StringType,
+		Token: "my:enum",
+		Annotations: []interface{}{
+			enumSchemaType{
+				Type: &schema.EnumType{Elements: []*schema.Enum{{Value: "foo"}, {Value: "bar"}}},
+			},
+		},
+	}
+	safeEnumFunc := func(member *schema.Enum) {}
+	unsafeEnumFunc := func(from model.Expression) {}
+
+	d := GenEnum(enum, &model.LiteralValueExpression{
+		Value: cty.StringVal("foo"),
+	}, safeEnumFunc, unsafeEnumFunc)
+	assert.Nil(t, d)
+
+	d = GenEnum(enum, &model.LiteralValueExpression{
+		Value: cty.StringVal("Bar"),
+	}, safeEnumFunc, unsafeEnumFunc)
+	assert.Equal(t, d.Summary, "\"Bar\" is not a valid value of the enum \"my:enum\"")
+	assert.Equal(t, d.Detail, "Valid members are \"foo\", \"bar\"")
```

Comment:
Go has raw string literals denoted with \`. It makes these look cleaner.
```suggestion
	assert.Equal(t, d.Summary, `"Bar" is not a valid value of the enum "my:enum"`)
	assert.Equal(t, d.Detail, `Valid members are "foo", "bar"`)
```